snap/gadget.go

```diff
@@ -103,6 +171,12 @@ func ReadGadgetInfo(info *Info, classic bool) (*GadgetInfo, error) {
 		gi.Defaults[k] = dflt.(map[string]interface{})
 	}
 
+	for _, gconn := range gi.Connects {
+		if (gconn == GadgetConnect{}) {
```

Comment:
I feel like having a `func (gconn *GadgetConnect) Empty() bool {..}`
member for tests like this is more appropriate and future-proof in case the struct gets any new member atrributes in the future as the function can make correct decision regardless of default values.